spynnaker/pyNN/models/neural_projections/projection_application_edge.py

```diff
@@ -109,6 +124,19 @@ def delay_edge(self):
     def delay_edge(self, delay_edge):
         self.__delay_edge = delay_edge
 
+    def is_neuromodulated(self, post_vertex):
+        """ Check if neuromodulation is set on any edge into the post_vertex
+
+        :rtype: bool
+        """
+        if self.__is_neuromodulated is None:
+            for proj in post_vertex.incoming_projections:
+                dynamics = proj._synapse_information.synapse_dynamics
+                if are_dynamics_stdp(dynamics):
+                    self.__is_neuromodulated = dynamics.neuromodulation
```

Comment:
There aren't currently any examples with a neuromodulated and non-neuromodulated STDP dynamics on the same edge.  I suppose there may be future cases where this is the case though, so you're probably right that this needs to stay true in a mixed case.

Reply:
In fact, given some of the sensible comments further on in this PR, it would probably be a good idea to write some tests where there are mixed dynamics objects on the same edge in order to make sure nothing is broken in those cases...